File: problems/CSES/Sorting_and_Searching/apartments/apartments.py
import sys
# sys.setrecursionlimit(200000)

input = sys.stdin.readline

def main():
    n, m, k = map(int, input().split())
    list_a = sorted(list(map(int, input().split())))
    list_b = sorted(list(map(int, input().split())))

    res = 0
    i = j = 0
    while i < n and j < m:
        if list_b[j] < list_a[i] - k:
            j += 1
        elif list_b[j] > list_a[i] + k:
            i += 1
        else:
            res += 1
            i += 1
            j += 1

    print(res)

main()

File: problems/CSES/Sorting_and_Searching/apartments/apartments.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

void solve() {
    return;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int n, m, k;
    cin >> n >> m >> k;

    vector<int> list_a(n);
    vector<int> list_b(m);

    for (int i = 0; i < n; i++) {
        cin >> list_a[i];
    }

    for (int i = 0; i < m; i++) {
        cin >> list_b[i];
    }

    sort(list_a.begin(), list_a.end());
    sort(list_b.begin(), list_b.end());

    int res = 0;
    int i = 0, j = 0;

    while (i < n && j < m) {
        if (list_b[j] < list_a[i] - k) {
            j += 1;
        } else if (list_b[j] > list_a[i] + k) {
            i += 1;
        } else {
            res += 1;
            i += 1;
            j += 1;
        }
    }

    cout << res << "\n";
    
    return 0;
}
